smart_open/smart_open_lib.py

```diff
@@ -577,6 +580,8 @@ def _compression_wrapper(file_obj, filename, mode):
         return BZ2File(file_obj, mode)
     elif ext == '.gz':
         return gzip.GzipFile(fileobj=file_obj, mode=mode)
+    elif ext == '.xz' and not IS_PY2:
```

Comment:
Do we want smart_open to behave differently on the same file type, across py2 / py3? (decompress in py3, leave compressed in py2)

That could generate nasty surprises for users, which I don't like.

Reply:
We can try to import pyLZMA on PY2, but if we fail, we don't have any other option besides writing a RuntimeWarning. WDYT?